src/data-access/repositories/groupRepository.ts

```diff
@@ -0,0 +1,79 @@
+import { DbError } from '../../core/errors/dbError.js';
+import { BaseGroup, Group } from '../../types/group.js';
+import { UserModel } from '../models/userModel.js';
+import { InitializeSequelize } from '../../database/postgreSQL/initializeSequelize.js';
+import { Model, Transaction } from 'sequelize';
+import { BaseRepository } from './baseRepository.js';
+
+
+export class GroupRepository extends BaseRepository {
+    constructor(groupModel: any, groupDataMapper: any) {
+        super(groupModel, groupDataMapper);
+        this.model = groupModel;
+        this.dataMapper = groupDataMapper;
+    }
+
+    async get(id: string): Promise<Group | undefined> {
+        const groupFromDB = await super.getEntityById(id);
+        return groupFromDB ? this.dataMapper.toDomain(groupFromDB.toJSON()) : undefined;
+    }
+
+    async getAll(): Promise<Array<Group | undefined>> {
+        let groupsFromDB;
+        try {
+            groupsFromDB = await this.model.findAll({
+                include: [
+                    {
+                        model: UserModel,
+                        as: 'users',
+                        attributes: ['id', 'login', 'password', 'age'],
+                        through: {
+                            attributes: []
+                        }
+                    }
+                ]
+            });
+        } catch (e) {
+            throw new DbError('Error retrieving group');
+        }
+        return groupsFromDB.map((group: Model) => {
+            return this.dataMapper.toDomain(group.toJSON());
+        });
+    }
+    async create(group: Group): Promise<Group> {
+        const groupToCreate = this.dataMapper.toDalEntity(group);
+        const createdGroup = await super.createEntity(groupToCreate);
+        return this.dataMapper.toDomain(createdGroup.toJSON());
+    }
+    async update(groupUpdates: BaseGroup, id: string): Promise<Group> {
```

Comment:
the same is here, pay your attention on what you pass into the method and what you get from it.